test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,672 @@
+        testutil_assert(strcmp(range_key, normal_key) == 0);
+        while (true) {
+            if (next) {
+                normal_ret = normal_cursor->next(normal_cursor.get());
+                range_ret = range_cursor->next(range_cursor.get());
+            } else {
+                normal_ret = normal_cursor->prev(normal_cursor.get());
+                range_ret = range_cursor->prev(range_cursor.get());
+            }
+            testutil_assert(normal_ret == 0 || normal_ret == WT_NOTFOUND);
+            testutil_assert(range_ret == 0 || range_ret == WT_NOTFOUND);
+
+            /* Early exit if we have reached the end of the table. */
+            if (range_ret == WT_NOTFOUND && normal_ret == WT_NOTFOUND)
+                break;
+
+            /* It is possible that we have reached the end of the bounded range. */
+            else if (range_ret == WT_NOTFOUND && normal_ret == 0) {
+                testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+                std::string normal_key_str = normal_key;
+                /*  Make sure that normal cursor returns a key that is outside of the range. */
+                if (next) {
+                    testutil_assert(upper_bound.key.length() != 0);
+                    testutil_assert(custom_lexicographical_compare(
+                                      normal_key_str, upper_bound.key, true) == false);
+                } else {
+                    testutil_assert(lower_bound.key.length() != 0);
+                    testutil_assert(custom_lexicographical_compare(
+                                      normal_key_str, lower_bound.key, false) == true);
+                }
+                break;
+            }
+
+            std::string range_key_str = range_key;
+            if (next && upper_bound.key.length() != 0)
+                testutil_assert(custom_lexicographical_compare(
+                                  range_key_str, upper_bound.key, upper_bound.inclusive) == true);
+            else if (!next && lower_bound.key.length() != 0)
+                testutil_assert(custom_lexicographical_compare(
+                                  lower_bound.key, range_key_str, lower_bound.inclusive) == true);
+            /* Make sure that records match between both cursors. */
+            testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+            testutil_check(range_cursor->get_key(range_cursor.get(), &range_key));
+            testutil_assert(strcmp(range_key, normal_key) == 0);
+        }
+    }
+
+    /*
+     * Use the random generator either set no bounds, only lower bounds, only upper bounds or both
+     * bounds on the range cursor. The lower and upper bounds are randomly generated strings as
+     * well. The inclusive configuration is also randomly set as well.
+     */
+    std::pair<bound, bound>
+    set_random_bounds(test_harness::thread_context *tc, scoped_cursor &range_cursor)
```

Comment:
Do we need `test_harness::` while you are in the namespace? I always get confused.